### ISSUE TITLE
[MLv2] No way to distinguish multiple columns with same name in Order By

### ISSUE THREAD

**Issue by camsaul** [metabase/metabase#29731]
Which `Created At` is which?

![image](https://user-images.githubusercontent.com/1455846/229201730-7b106ab4-bc4c-42d9-b87c-a74d53927a20.png)

This is something we can fix as part of MLv2.

**Comment by snoe** [comment]
Should (may already) be fixed by using `long-display-name` https://github.com/metabase/metabase/pull/30882

**Comment by darksciencebase** [comment]
It got longer but not more different, I think. This is on a local `1f7e261ec`
![image](https://github.com/metabase/metabase/assets/125455699/dc674db5-f0f6-466a-9848-84b8ca49f607)

**Comment by metamben** [comment]
This is working (almost) properly on the #30693 branch. The only problem seems to be that the `QueryColumnPicker` component uses the ordinary `displayName` property instead of `longDisplayName`.

I leave it to @kulyk to decide what the best fix for this is, as just changing the property might break other parts of the editor.

**Comment by kulyk** [comment]
Will be fixed once #30693 is merged. LMK if I should pull it into a separate PR

**Comment by metamben** [comment]
I don't think a separate PR is needed.
